junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestExtension.java

```diff
@@ -91,6 +92,19 @@ public boolean supportsTestTemplate(ExtensionContext context) {
 		// @formatter:on
 	}
 
+	@SuppressWarnings("ConstantConditions")
+	private ArgumentsProvider getArgumentsProvider(Class<? extends ArgumentsProvider> clazz) {
+		try {
+			return ReflectionUtils.newInstance(clazz);
+		}
+		catch (Exception e) {
+			if (e instanceof NoSuchMethodException) {
```

Comment:
No need to perform an `instanceof` check. Simply catch a `NoSuchMethodException` and otherwise allow the exception to propagate as is.

Reply:
`Exception 'java.lang.NoSuchMethodException' is never thrown in the corresponding try block` since `ReflectionUtils#newInstance` catches and throws a `RuntimeException`.

Reply:
Ahhh.... OK. Sorry.